### PR TITLE
fix some ineffassign issues

### PR DESCRIPTION
**Pull request by vbehar**
reported by https://goreportcard.com/report/github.com/openshift/origin#ineffassign

mostly shadowing issues

### CODE REVIEW

File: pkg/client/localsubjectaccessreview.go [hunk 1]
```diff
@@ -63,7 +63,7 @@ func (c *localSubjectAccessReviews) Create(sar *authorizationapi.LocalSubjectAcc
 		}
 		deprecatedResponse := &authorizationapi.SubjectAccessReviewResponse{}
 
-		req, err := overrideAuth(c.token, c.r.Post().Namespace(c.ns).Resource("subjectAccessReviews"))
+		req, err = overrideAuth(c.token, c.r.Post().Namespace(c.ns).Resource("subjectAccessReviews"))
```
Comment (deads2k) [inline pkg/client/localsubjectaccessreview.go]: So I thought this created a new `req` variable in this scope instead of re-using the one from above.  I don't like re-using variables.  If we want to remove the marker, you could change it to a different variable name?

Comment (kargakis) [inline pkg/client/localsubjectaccessreview.go]: Agreed, let's use a different name.

Comment (vbehar) [inline pkg/client/localsubjectaccessreview.go]: ok, I renamed it to deprecatedReq (to be consistent with deprecatedResponse)

File: pkg/client/localsubjectaccessreview.go [hunk 2]
```diff
@@ -63,11 +64,12 @@ func (c *localSubjectAccessReviews) Create(sar *authorizationapi.LocalSubjectAcc
 		}
 		deprecatedResponse := &authorizationapi.SubjectAccessReviewResponse{}
 
-		req, err := overrideAuth(c.token, c.r.Post().Namespace(c.ns).Resource("subjectAccessReviews"))
+		var deprecatedReq *restclient.Request
```
Comment (kargakis) [inline pkg/client/localsubjectaccessreview.go]: Is there any reason to declare now that you renamed it?

File: pkg/client/subjectaccessreview.go [hunk 3]
```diff
@@ -64,7 +64,8 @@ func (c *subjectAccessReviews) Create(sar *authorizationapi.SubjectAccessReview)
 	// if the namespace values don't match then we definitely hit an old server.  If we got a forbidden, then we might have hit an old server
 	// and should try the old endpoint
 	if (sar.Action.Namespace != result.Namespace) || kapierrors.IsForbidden(err) {
-		req, err := overrideAuth(c.token, c.r.Post().Namespace(sar.Action.Namespace).Resource("subjectAccessReviews"))
+		var req *restclient.Request
```
Comment (kargakis) [inline pkg/client/subjectaccessreview.go]: Why this change?

File: pkg/client/localsubjectaccessreview.go (hunk 2 above)
Comment (vbehar) [inline pkg/client/localsubjectaccessreview.go]: yes, it is to avoid using `:=` to initialize the vars, and so shadow the `err` var, which could then be set to `nil`

File: pkg/client/subjectaccessreview.go (hunk 3 above)
Comment (vbehar) [inline pkg/client/subjectaccessreview.go]: same as the other change : to avoid using `:=` to initialize the vars, and so shadow the `err` var, which could then be set to `nil`

File: pkg/client/localsubjectaccessreview.go (hunk 2 above)
Comment (soltysh) [inline pkg/client/localsubjectaccessreview.go]: But we don't want to change the previous `err` value, do we? At least not at this point. Only line 74 should do that. I'd suggest using different name for that error.

File: pkg/client/subjectaccessreview.go (hunk 3 above)
Comment (soltysh) [inline pkg/client/subjectaccessreview.go]: My worries from above apply here as well.

File: pkg/client/localsubjectaccessreview.go (hunk 2 above)
Comment (vbehar) [inline pkg/client/localsubjectaccessreview.go]: yes you are correct. I have re-done it with a different name for the second error, it is clearer.

**Comment by soltysh** [comment]
LGTM now, @kargakis the merge is yours, though.

**Comment by kargakis** [comment]
LGTM [merge]
